### PR TITLE
Multipart form data #1276

### PR DESCRIPTION
**Pull request by timtjtim**
I'd like to add some more error cases, but that can be on another issue.

### CODE REVIEW

File: src/handle_form.inl [hunk 1]
```diff
@@ -701,43 +705,71 @@ mg_handle_form_request(struct mg_connection *conn,
...
+			 */
+
 			if (part_no == 0) {
-				int d = 0;
-				while ((d < buf_fill) && (buf[d] != '-')) {
-					d++;
+				size_t preamble_length = 0;
+				/* skip over the preamble until we find a complete boundary */
+				/* +2 for the -- preceding the boundary */
+				while ((preamble_length < buf_fill - bl)
```
Comment (timtjtim) [inline src/handle_form.inl:721]: Could drop this / add a preamble length limit if we're worried about legitimate request sizes

Comment (bel2125) [reply src/handle_form.inl:721]: I have no idea why someone would want to use a preamble at all.
A preamble >4kB seems excessive and is probably only sent to block the server.

**Comment by bel2125** [comment]
starting CI tests ... this may take an hour.

File: unittest/public_server.c [hunk 2]
```diff
@@ -57,6 +57,79 @@
 #define SLEEP_AFTER_MG_STOP (5)
 
 
+#ifdef HEXDUMP
+#else
+void
+hexDumpth(const char *desc, const void *addr, const int len, int perLine)
```
Comment (timtjtim) [inline unittest/public_server.c:63]: Oh - didn't mean to commit this debug method

File: src/handle_form.inl (hunk 1 above)
Comment (timtjtim) [reply src/handle_form.inl]: Done - went for 1024

**Comment by bel2125** [comment]
Thank you.
Merged.

**Pull request merged**